### PR TITLE
Add Type Definitions for TypeScript

### PR DESCRIPTION
**Pull request by proudust**

### CODE REVIEW

**Comment by sibiraj-s** [comment]
@proudust Thanks for the PR. I will try to merge it this weekend.

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import type { ArchiverOptions } from 'archiver';
```
Comment (sibiraj-s) [inline types.d.ts:1]: I don't think archiver has published types. we may need to add a dependency `@types/archiver`

**Review by sibiraj-s — CHANGES_REQUESTED** [review]

File: types.d.ts (hunk 1 above)
Comment (proudust) [reply types.d.ts:1]: Added to devDependencies.

Comment (sibiraj-s) [reply types.d.ts:1]: It should be a dependency. the package manager won't install devDependencies during `npm/yarn install <package>`

**Review by sibiraj-s — APPROVED** [review]

**Comment by sibiraj-s** [comment]
Thanks @proudust . Published in v6.1.5

**Pull request merged**